File: DemoMMOUnity/Assets/Game/Net/Script/Net/CsProtocol/Protocol/Map/PuzzleAllConfigRequest.cs
using Newtonsoft.Json;
using ZJYFrameWork.Collection.Reference;
using ZJYFrameWork.Net.Core;

namespace ZJYFrameWork.Net.CsProtocol.Buffer.Protocol.Map
{
    /// <summary>
    /// 地图 所有总信息 request
    /// </summary>
    public class PuzzleAllConfigRequest : Model, IPacket, IReference
    {
        /// <summary>
        /// 事件id 也可以
        /// </summary>
        public int eventId { get; set; }

        public short ProtocolId()
        {
            return 1035;
        }

        public void Clear()
        {
            eventId = 0;
        }

        public static PuzzleAllConfigRequest ValueOf()
        {
            var packet = ReferenceCache.Acquire<PuzzleAllConfigRequest>();
            packet.Clear();
            return packet;
        }
    }

    public class PuzzleAllConfigRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1035;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            if (packet == null)
            {
                return;
            }

            var data = (PuzzleAllConfigRequest)packet;
            var packetData = new ServerMessageWrite(ProtocolId(), data);
            var jsonString = JsonConvert.SerializeObject(packetData);
            buffer.WriteString(jsonString);
        }

        public IPacket Read(string json = "")
        {
            var packet = PuzzleAllConfigRequest.ValueOf();
            if (string.IsNullOrEmpty(json))
            {
                return packet;
            }

            return packet;
        }
    }
}

File: UnityMMOClient/Lua/application/app/net/webSocket/luaProtocol/puzzle/puzzleAllConfigRequest.lua
---@class PuzzleAllConfigRequest
local PuzzleAllConfigRequest = class("PuzzleAllConfigRequest")
function PuzzleAllConfigRequest:ctor()
    --- 事件ID 也可以说是 活动id
    ---@type number
    self.eventId = 0
end

---@param eventId number 事件ID 也可以说是 活动id
---@return PuzzleAllConfigRequest
function PuzzleAllConfigRequest:new(eventId)
    self.eventId = eventId --- int
    return self
end

---@return number
function PuzzleAllConfigRequest:protocolId()
    return 1035
end

---@return string
function PuzzleAllConfigRequest:write()
    local message = {
        protocolId = self:protocolId(),
        packet = {
            eventId = self.eventId
        }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

---@return PuzzleAllConfigRequest
function PuzzleAllConfigRequest:read(data)

    local packet = self:new(
            data.eventId)
    return packet
end

--- 事件ID 也可以说是 活动id
---@return number 事件ID 也可以说是 活动id
function PuzzleAllConfigRequest:getEventId()
    return self.eventId
end


return PuzzleAllConfigRequest
